lib/foods.js

```diff
@@ -32,47 +33,59 @@ $('document').ready(function() {
   })
 
   $('tbody').on("click", ".food-delete", function() {
-    // console.log('clicked');
-    $(this).parent().remove();
+    $(this).parent().addClass("hidden");
+    var name = $(this).siblings(".food-name").text();
+    var foodToHide = _.find(foodList, function(food) {
+      return food.name == name
+    });
+    foodToHide.visibility = "hidden"
+    localStorage.setItem('foods', JSON.stringify(foodList));
   })
 
-  $("#food-filter input").keyup(function () {
-    //split the current value of the search input
-    var searchInput = this.value.toUpperCase().split(" ");
-    //create a jquery object of the rows
-    var rows = $("#food-items").find("tr");
-    if (this.value == "") {
-      rows.show();
-      return;
-    }
-    //hide all the rows
-    rows.hide();
+  $("#food-list").on("focus", ".food-name", function() {
+    originalName = $(this).text();
+  })
 
-    //Recusively filter the jquery object to get results.
-    rows.filter(function () {
-      for (var i = 0; i < searchInput.length; ++i) {
-        if ($(this).text().toUpperCase().indexOf(searchInput[i]) > -1) {
-          return true;
-        }
-      }
-      return false;
-    })
-    //show the rows that match.
-    .show();
+  $("#food-list").on("focus", ".food-calories", function() {
+    originalName = $(this).siblings(".food-name").text();
   })
 
   $("#food-list").on('keydown', ".food-name, .food-calories", function(event){
     if (event.keyCode == 13) {
+      var foodToUpdate = _.find(foodList, function(food) {
+        return food.name == originalName
+      });
+      foodToUpdate.name =     $(this).parent().children(".food-name").text();
+      foodToUpdate.calories = $(this).parent().children(".food-calories").text();
+      localStorage.setItem('foods', JSON.stringify(foodList));
       event.preventDefault();
       this.blur();
     }
   })
 
-  function addFoodToTable(name, calories) {
+  $("#food-list").on('blur', ".food-name, .food-calories", function(event) {
+    var foodToUpdate = _.find(foodList, function(food) {
+      return food.name == originalName
+    });
+    foodToUpdate.name =     $(this).parent().children(".food-name").text();
+    foodToUpdate.calories = $(this).parent().children(".food-calories").text();
+    localStorage.setItem('foods', JSON.stringify(foodList));
+  });
+
+  function updateNameAndCalories() {
```

Comment:
This looks like dead code... it looks like you were trying to extract out similar functionality to lines 58-60 here. Something to revisit.